src/nvim/os/stdpaths.c

```diff
@@ -188,6 +212,9 @@ char *stdpaths_get_xdg_var(const XDGVarType idx)
     ret = xdg_remove_duplicate(ret, ENV_SEPSTR);
   }
 
+  if (memfree) {
```

Comment:
could avoid the need for `memfree` by changing line 192 to:

    env_val = xstrdup("");